core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileSourceConfig.java

```diff
@@ -31,196 +32,59 @@
-  @VisibleForTesting
-  static final long DEFAULT_MAX_SPLIT_SIZE = 134217728;
 
-  @Nullable
-  @Description(FILESYSTEM_PROPERTIES_DESCRIPTION)
-  @Macro
-  public String fileSystemProperties;
-
-  @Nullable
-  @Description(REGEX_DESCRIPTION)
-  @Macro
-  public String fileRegex;
-
-  @Nullable
-  @Description(TABLE_DESCRIPTION)
   @Macro
-  public String timeTable;
+  @Description("Path to file(s) to be read. If a directory is specified, " +
+    "terminate the path name with a \'/\'. For distributed file system such as HDFS, file system name should come"  +
+    "from 'fs.DefaultFS' property in the 'core-site.xml'. For example, 'hdfs://mycluster.net:8020/input', where " +
+    "value of the property 'fs.DefaultFS' in the 'core-site.xml' is 'hdfs://mycluster.net:8020'.")
+  private String path;
 
-  @Nullable
-  @Description(INPUT_FORMAT_CLASS_DESCRIPTION)
   @Macro
-  public String inputFormatClass;
-
   @Nullable
-  @Description(FORMAT_DESCRIPTION)
-  public String format;
+  @Description("Any additional properties to use when reading from the filesystem. "
+    + "This is an advanced feature that requires knowledge of the properties supported by the underlying filesystem.")
+  private String fileSystemProperties;
 
-  @Nullable
-  @Description(MAX_SPLIT_SIZE_DESCRIPTION)
   @Macro
-  public Long maxSplitSize;
-
-  @Nullable
-  @Description("Identify if path needs to be ignored or not, for case when directory or file does not exists. If " +
-    "set to true it will treat the not present folder as zero input and log a warning. Default is false.")
-  public Boolean ignoreNonExistingFolders;
-
-  @Nullable
-  @Description("Boolean value to determine if files are to be read recursively from the path. Default is false.")
-  public Boolean recursive;
-
-  @Nullable
-  @Description("If specified, each output record will include a field with this name that contains the file URI " +
-    "that the record was read from. Requires a customized version of CombineFileInputFormat, so it cannot be used " +
-    "if an inputFormatClass is given.")
-  public String pathField;
-
-  @Nullable
-  @Description("If true and a pathField is specified, only the filename will be used. If false, the full " +
-    "URI will be used. Defaults to false.")
-  public Boolean filenameOnly;
-
   @Nullable
-  @Description("Schema for the source")
-  public String schema;
+  @Description("Deprecated property. Use the logicalStartTime macro in the file path instead of this.")
+  private String timeTable;
 
-  // this is a hidden property that only exists for wrangler's parse-as-csv that uses the header as the schema
-  // when this is true and the format is text, the header will be the first record returned by every record reader
-  @Nullable
-  private Boolean copyHeader;
-
-  public FileSourceConfig() {
-    this(null, null, null, null, null, null, null, null, null, null, null, null);
+  FileSourceConfig() {
+    super();
+    fileSystemProperties = "{}";
```

Comment:
why this is not null? The removed code assigns null and it also has a check for it in `getFileSystemProperties`